### ISSUE TITLE
知识蒸馏

### ISSUE THREAD

**Issue by zys1994** [changewOw/jishi-fashion-classifier-comp-2nd#1]
关于知识蒸馏的开源工程可以分享一下吗

**Comment by hammer422** [comment]
1.自我蒸馏模式(我的比赛最终选择方案):
**e4.py** 会进行5折训练，然后用这5折分别对他们的验证集进行预测并保存预测结果 `torch.save(results, "/project/train/models/v1_ema/oof.pth")`   oof.pth实际是个字典保存了image_name和预测的label
**m11.py** 在dataset里面
 ` label_np = np.zeros((config.num_classes), dtype=np.float32)
 label_np[label] += 0.8
 label_np[oof_p[image_name]] += 0.2
`

2.teacher-student蒸馏模式:我参照的项目是 [KD](https://github.com/peterliht/knowledge-distillation-pytorch)
**e4.py** 生成训练好的大模型(efficientnet-b1 or efficientnet-b4)就是模型文件
**m11kd.py**(刚更新0619/13:00)去训练mobilenetV3-small， 与第一种不同的地方在于

1.  [生成teacher模型](https://github.com/changewOw/jishi-fashion-classifier-comp-2nd/blob/master/m11kd.py#L1559)
2. [训练生成teacher的logits](https://github.com/changewOw/jishi-fashion-classifier-comp-2nd/blob/master/m11kd.py#L1478)
3. [KD-loss 我试的最好的参数是T=20, alpha=0.9, 小幅度变化都没问题](https://github.com/changewOw/jishi-fashion-classifier-comp-2nd/blob/master/m11kd.py#L1354)

其他都无太大差别,唯一要注意的就是我的e4.py是按第一种模式跑的，走第二种模式需要注意teacher和student的数据增强最好保持一致.

**Comment by zys1994** [comment]
非常感谢，数据增强这一块我感觉可以用Imgaug进一步增强
